images/dcos/bootstrap.sh

```diff
@@ -163,6 +154,32 @@ EOF
   prepare_service_depends etcd-server ${etcd_server_list}/v2/stats/store getsSuccess ${deps}
 }
 
+prepare_etcd_proxy() {
+  prepare_service ${monitor_dir} ${service_dir} etcd-server ${ETCD_SERVER_RESPAWN_DELAY:-1} << EOF
+#!/bin/sh
+#TODO(jdef) don't run this as root
+#TODO(jdef) would be super-cool to have socket-activation here so that clients can connect before etcd is really ready
+exec 2>&1
+mkdir -p ${etcd_server_data}
+PATH="/opt:${PATH}"
+export PATH
+exec /opt/etcd \\
+  -proxy=on \\
+  -advertise-client-urls ${etcd_advertise_client_urls} \\
+  -data-dir ${etcd_server_data} \\
+  -listen-client-urls ${etcd_listen_client_urls} \\
+  -discovery-srv=${ETCD_MESOS}.mesos \\
+EOF
+
+  local deps="controller-manager scheduler"
```

Comment:
these deps were copied from prepare_etcd_server. at this point the controller manager should not have a direct dependency to etcd so we should be safe to remove it from both this script and prepare_etcd_server.